xarray/core/combine.py

```diff
@@ -605,21 +560,141 @@ def auto_combine(datasets, concat_dim=_CONCAT_DIM_DEFAULT,
     --------
     concat
     Dataset.merge
-    """  # noqa
-
-    # Coerce 1D input into ND to maintain backwards-compatible API until API
-    # for N-D combine decided
-    # (see https://github.com/pydata/xarray/pull/2553/#issuecomment-445892746)
-    if concat_dim is None or concat_dim == _CONCAT_DIM_DEFAULT:
-        concat_dims = concat_dim
-    elif not isinstance(concat_dim, list):
-        concat_dims = [concat_dim]
+    """
+
+    if concat_dim is '_not_supplied':
+        concat_dim = _CONCAT_DIM_DEFAULT
+    else:
+        message = """In xarray version 0.14 `auto_combine` will be deprecated,
+                  and `open_mfdataset` will no longer accept a `concat_dim`
+                  argument. To get equivalent behaviour from now on please use
+                  the new `combine_manual` function instead (or the
+                  `combine='manual'` option to open_mfdataset)."""
+        warnings.warn(message, FutureWarning)
```

Comment:
Please set `stacklevel` when you call `warnings.warn()` so users get an error message where they used `xarray.auto_combine()` in their code. In most cases, I think the right level is `stacklevel=2` or maybe stacklevel=3`, but you should experiment a little to make sure.